composition-js/src/__tests__/hints.test.ts

```diff
@@ -603,4 +606,79 @@ test('hints on inconsistent description for field', () => {
     + '  I don\'t know what I\'m doing\n'
     + '  """'
   );
-})
+});
+
+describe('hint tests related to the @override directive', () => {
```

Comment:
All the hints currently raises are field based. So in practice, if you were to move a type having 50 fields to another subgraph using a type-level `@override`, we'll generate 50 hints, which isn't an amazing user experience. Of course, generating type-level hints for this is also a bit tricky, because even if you put a `@override` on a type, the hint generated for each field of that type might actually differ in some cases (maybe the overriding type has actually more fields than the overridden one for instance). But I think overwhelming users with hints is a genuine potential problem and worth expanding some effort to improve.

That said, I'm ok punting on that problem in this PR, because we already know we need to rethink and cleanup the experience with hints and so we can defer the problem here then. But wanted to mention it nonetheless so that 1) we collectively have it in mind and 2) in case you actually prefer tackling that problem now.

Reply:
Ok, I agree and think that we should punt as it seems like we might need a more general solution. Thanks for thinking about this.